plugin/src/main/kotlin/tanvd/grazi/language/Lang.kt

```diff
@@ -1,47 +1,59 @@
-    RUSSIAN(Russian(), GraziBundle.langConfig("ru.enabled")),
-    PERSIAN(Persian()),
-    FRENCH(French()),
-    GERMANY_GERMAN(GermanyGerman()),
-    AUSTRIAN_GERMAN(AustrianGerman()),
-    POLISH(Polish()),
-    ITALIAN(Italian()),
-    DUTCH(Dutch()),
-    PORTUGAL_PORTUGUESE(PortugalPortuguese()),
-    BRAZILIAN_PORTUGUESE(BrazilianPortuguese()),
-    CHINESE(Chinese()),
-    GREEK(Greek()),
-    JAPANESE(Japanese()),
-    ROMANIAN(Romanian()),
-    SLOVAK(Slovak()),
-    SPANISH(Spanish()),
-    UKRAINIAN(Ukrainian());
+                private val disabledRules: Set<String> = emptySet(),
+                private val disabledCategories: Set<String> = emptySet()) {
+    BRITISH_ENGLISH("English (GB)", "en", "BritishEnglish", "14 MB",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    AMERICAN_ENGLISH("English (US)", "en", "AmericanEnglish", "14 MB",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    CANADIAN_ENGLISH("English (Canadian)", "en", "CanadianEnglish", "14 MB",
+            GraziBundle.langConfig("en.rules.enabled"), GraziBundle.langConfig("en.rules.disabled"), GraziBundle.langConfig("en.categories.disabled")),
+    RUSSIAN("Russian", "ru", "Russian", "3 MB", GraziBundle.langConfig("ru.rules.enabled")),
+    PERSIAN("Persian", "fa", "Persian", "1 MB"),
+    FRENCH("French", "fr", "French", "4 MB"),
+    GERMANY_GERMAN("German (Germany)", "de", "GermanyGerman", "19 MB"),
+    AUSTRIAN_GERMAN("German (Austria)", "de", "AustrianGerman", "19 MB"),
+    POLISH("Polish", "pl", "Polish", "5 MB"),
+    ITALIAN("Italian", "it", "Italian", "1 MB"),
+    DUTCH("Dutch", "nl", "Dutch", "17 MB"),
+    PORTUGAL_PORTUGUESE("Portuguese (Portugal)", "pt", "PortugalPortuguese", "5 MB"),
+    BRAZILIAN_PORTUGUESE("Portuguese (Brazil)", "pt", "BrazilianPortuguese", "5 MB"),
+    CHINESE("Chinese", "zh", "Chinese", "3 MB"),
+    GREEK("Greek", "el", "Greek", "1 MB"),
+    JAPANESE("Japanese", "ja", "Japanese", "1 MB"),
+    ROMANIAN("Romanian", "ro", "Romanian", "1 MB"),
+    SLOVAK("Slovak", "sk", "Slovak", "3 MB"),
+    SPANISH("Spanish", "es", "Spanish", "2 MB"),
+    UKRAINIAN("Ukrainian", "uk", "Ukrainian", "6 MB");
 
     companion object {
-        operator fun get(lang: Language): Lang? = values().find { it.shortCode == lang.shortCode }
+        operator fun get(lang: Language): Lang? = values().find { lang.name == it.displayName }
         operator fun get(code: String): Lang? = values().find { it.shortCode == code }
 
-        val sortedValues = values().sortedBy { it.displayName }
+        fun sortedValues() = values().sortedBy(Lang::displayName)
     }
 
-    val shortCode = jLanguage.shortCode!!
+    private var _jLanguage: Language? = null
+    val jLanguage: Language?
+        get() {
+            if (_jLanguage == null) {
```

Comment:
As a fun excersise I would recommend implementing `lazyNotNull`, but ok)

Reply:
Lazy not that simple, standard lazy also cares about threadsafety and have three different types to use :) So, probably later.